src/lib/snark_work_lib/selector.ml

```diff
@@ -0,0 +1,82 @@
+open Core_kernel
+
+module Single = struct
+  module Spec = struct
+    [%%versioned
+    module Stable = struct
+      [@@@no_toplevel_latest_type]
+
+      module V1 = struct
+        type t =
+          ( Transaction_witness.Stable.V2.t
+          , Ledger_proof.Stable.V2.t )
+          Work.Single.Spec.Stable.V2.t
+        [@@deriving sexp, yojson]
```

Comment:
Do we need `of_yojson`, `sexp_of`?

Reply:
If it's only for logging, we could use `to_yojson` instead of `yojson`. If that would work, if not, it means it isn't only for logging :D

Reply:
Let me check if that's doable :)

Reply:
I experience build failure just now. Some how dune build didn't figured out the first time I'm implementing this. 

It's here requiring `of_yojson`, should I delete reading sexp? 

https://github.com/MinaProtocol/mina/blob/0c4bd2669ca4b7c3c1e2e645a775ef36737738c0/src/lib/snark_worker/standalone/run_snark_worker.ml#L91

BTW, I think we should drop either sexp or json, and only retain one serialization if possible.